dfa/src/main/scala/dfa/DFA.scala

```diff
@@ -1,3 +1,23 @@
 package dfa // leave this line in the file
 
 // TODO: replace this comment with your implementation
+
+case class State (label:String)
+case class Transition (from:State, to:State, symbol:Char)
+
+class DFA(states:Set[State], val transitions:Set[Transition],
+val start: State, val accept:Set[State]):
+    
+  def accepts(input:String): Boolean = {
+    input match {
+    case "" => 
+      if (accept.contains(this.start)){return true} 
+      else {return false}
+    case stringInput:Any =>
+      val newList = this.transitions.filter(_.from == this.start).filter(_.symbol == stringInput.charAt(0))
```

Comment:
Notably, the sample solution throws an exception when it cannot find a valid transition. I have not thought about doing so in my code either, but it is especially a good idea when the DFA inputs become long and computationally expensive.